tools/rulegen/grpc_gateway.go

```diff
@@ -79,40 +76,40 @@ var grpcGatewayLibraryExampleTemplate = mustTemplate(`load("@rules_proto_grpc//{
 
 func makeGrpcGateway() *Language {
 	return &Language{
-		Dir:  "github.com/grpc-ecosystem/grpc-gateway",
-		Name: "grpc-gateway",
+		Dir:         "github.com/grpc-ecosystem/grpc-gateway",
+		Name:        "grpc-gateway",
 		DisplayName: "grpc-gateway",
-		Flags: commonLangFlags,
+		Flags:       commonLangFlags,
 		Rules: []*Rule{
 			&Rule{
 				Name:             "gateway_grpc_compile",
 				Kind:             "grpc",
 				Implementation:   aspectRuleTemplate,
-				Plugins:          []string{"//github.com/grpc-ecosystem/grpc-gateway:grpc_gateway_plugin", "//go:grpc_go_plugin"},
+				Plugins:          []string{"//github.com/grpc-ecosystem/grpc-gateway:grpc_gateway_plugin", "//go:grpc_go_plugin", "//go:go_plugin"},
 				WorkspaceExample: grpcGatewayWorkspaceTemplate,
 				BuildExample:     grpcGatewayCompileExampleTemplate,
 				Doc:              "Generates grpc-gateway `.go` files",
 				Attrs:            aspectProtoCompileAttrs,
 			},
 			&Rule{
-				Name:             "gateway_swagger_compile",
-				Kind:             "grpc",
-				Implementation:   aspectRuleTemplate,
-				Plugins:          []string{"//github.com/grpc-ecosystem/grpc-gateway:swagger_plugin"},
-				WorkspaceExample: grpcGatewayWorkspaceTemplate,
-				BuildExample:     grpcGatewayCompileExampleTemplate,
-				Doc:              "Generates grpc-gateway swagger `.json` files",
-				Attrs:            aspectProtoCompileAttrs,
+				Name:              "gateway_openapiv2_compile",
+				Kind:              "grpc",
+				Implementation:    aspectRuleTemplate,
+				Plugins:           []string{"//github.com/grpc-ecosystem/grpc-gateway:openapiv2_plugin", "//go:go_plugin"},
```

Comment:
Why was `go_plugin` added for the JSON generation?

Reply:
I only vaguely remember - I think it was because the new `openapiv2_plugin` didn't generate the Go sources when used only by itself. It may be a wrong way to fix it, I'm not sure.